tests/test_text.py

```diff
@@ -0,0 +1,775 @@
+            {"item1": ["this is a sentences with more common words and more words to increase frequency"],
+             "item2": ["second one in list of more frequent sentences with some repeated words"],
+             "item3": ["a word for more complexity and longer length"],
+             "item4": ["sentence with a lot of repeated common words and more words to increase frequency"],
+             "item5": ["more frequent words with a valid vector and more words to increase frequency"]})
+        labels = pd.DataFrame({"item1": [0, 1, 0, 0, 0], "item2": [1, 0, 0, 1, 0], "item3": [0, 0, 1, 0, 1],
+                               "item4": [0, 1, 1, 0, 1], "item5": [0, 1, 0, 0, 1]})
+
+        # Verify that the number of columns is data and labels match
+        self.assertEqual(data.shape[1], labels.shape[1])
+
+        # Fix the random seed
+        seed = 123
+        np.random.seed(seed)
+
+        method_unicost = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="unicost",
+                                                   trials=1)
+
+        method_diverse = SelectionMethod.TextBased(num_features=3,
+                                                   optimization_method="random",
+                                                   cost_metric="diverse",
+                                                   trials=1)
+
+        # Find the best solution using multiple trails and assert that it is selected with the same seed
+        method = SelectionMethod.TextBased(num_features=3,
+                                           optimization_method="random",
+                                           cost_metric="unicost",
+                                           trials=20)
+
+        selector_unicost = Selective(method_unicost, seed=seed)
+        selector_unicost.fit(data, labels)
+        selected_features_unicost = selector_unicost.transform(data)
+
+        # Set the same seed again
+        np.random.seed(seed)
+        selector_diverse = Selective(method_diverse, seed=seed)
+        selector_diverse.fit(data, labels)
+        selected_features_diverse = selector_diverse.transform(data)
+
+        # Check whether the seed value used in tests is the same (not the default = 123456)
+        self.assertEqual(selector_unicost.seed, selector_diverse.seed)
+
+        # Verify the consistency of selected features with the initial run
+        self.assertTrue(selected_features_unicost.equals(selected_features_diverse))
+
+        # Set the same seed again
+        np.random.seed(seed)
+        best_selector = Selective(method, seed=seed)
+        best_selector.fit(data, labels)
+        best_selected_features = best_selector.transform(data)
+
+        # Verify the selected indices
+        self.assertListEqual(list(best_selector.transform(data).columns), ['item2', 'item3', 'item5'])
```

Comment:
did you mean to use "best_selected_features " from line 74 here?

Reply:
Yes, I edited it.

One other possible test could be to conduct four different tests, each with the same number of features and cost metrics, but with a different optimization method. For example, we could create a simulation for each optimization method, with 10 different seeds, and then find the mean and variance of the different optimization techniques. By comparing the results, we could suggest which method is most robust against randomness. Implementing this improvement might take some additional time, but if you're interested in the idea, I could work on it and provide the results.